openshift/default-catalog-consistency/pkg/check/filesystem.go

```diff
@@ -38,3 +41,67 @@ func FilesystemHasDirectories(paths ...string) FilesystemCheck {
 		},
 	}
 }
+
+// FilesystemHasValidPathsOrLinks checks that the given paths exist inside the extracted image filesystem.
+// It supports regular files and symlinks:
+//   - If it's a regular file, we just check that it exists.
+//   - If it's a symlink, we check that the link target exists.
```

Comment:
do we care if it points to a particular target or not? or only whether it exists?

Reply:
at the same time, this feels a bit like testing implementation - should we rather just do something like execute opm --version and grep for some string (or something like that). I.e. it shouldn't really matter if its a symlink or not, just that it can be executed and that its opm...?

Reply:
maybe doing something like: `go version -m bin/opm` ?